Add leaked credentials check feature

mrusso19 opened the pull request:
## Description

Add new methods to interact with the [Leaked Credentials Check Cloudflare API](https://developers.cloudflare.com/api/operations/waf-product-api-leaked-credentials-get-status)

## Types of changes

What sort of change does your code introduce/modify?

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

mrusso19 committed:
Add new error constant

mrusso19 committed:
Add API wrapper for leak credential detection

mrusso19 committed:
Add test for new API wrapper for leak credential check

mrusso19 committed:
Add changelog entry

jacobbednarz commented on leaked_credential_check.go:
```diff
@@ -0,0 +1,190 @@
+type LeakedCredentialCheckGetStatusParams struct{}
+
+type LeakedCredentialCheckStatus struct {
+	Enabled bool `json:"enabled"`
```
https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#booleans

```suggestion
	Enabled *bool `json:"enabled"`
```

jacobbednarz commented on leaked_credential_check.go:
```diff
@@ -0,0 +1,190 @@
+}
+
+type LeakCredentialCheckSetStatusParams struct {
+	Enabled bool `json:"enabled"`
```
https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#booleans

```suggestion
	Enabled *bool `json:"enabled"`
```

jacobbednarz commented on leaked_credential_check_test.go:
```diff
@@ -0,0 +1,190 @@
+	mux.HandleFunc("/zones/"+testZoneID+"/leaked-credential-checks", handler)
+
+	want := LeakedCredentialCheckStatus{
+		Enabled: true,
```
```suggestion
		Enabled: BoolPtr(true),
```

jacobbednarz commented on leaked_credential_check_test.go:
```diff
@@ -0,0 +1,190 @@
+	mux.HandleFunc("/zones/"+testZoneID+"/leaked-credential-checks", handler)
+
+	want := LeakedCredentialCheckStatus{
+		Enabled: false,
```
```suggestion
		Enabled: BoolPtr(false),
```

jacobbednarz committed:
swap booleans to pointers

https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#booleans

jacobbednarz committed:
Merge branch 'master' into add-leak-credential-check

jacobbednarz committed:
Merge branch 'master' into add-leak-credential-check

jacobbednarz committed:
Merge branch 'master' into add-leak-credential-check

jacobbednarz commented on leaked_credential_check_test.go:
```diff
@@ -0,0 +1,190 @@
+	want := LeakedCredentialCheckStatus{
+		Enabled: BoolPtr(false),
+	}
+	actual, err := client.LeakedCredentialCheckSetStatus(context.Background(), ZoneIdentifier(testZoneID), LeakCredentialCheckSetStatusParams{false})
```
```suggestion
	actual, err := client.LeakedCredentialCheckSetStatus(context.Background(), ZoneIdentifier(testZoneID), LeakCredentialCheckSetStatusParams{BoolPtr(false)})
```

jacobbednarz committed:
swap `false` to boolptr helper

jacobbednarz committed:
Merge branch 'master' into add-leak-credential-check

jacobbednarz committed:
Merge branch 'master' into add-leak-credential-check

jacobbednarz committed:
Merge branch 'master' into add-leak-credential-check
